packages/shared/src/hooks/useChat.ts

```diff
@@ -164,7 +168,12 @@ export const useChat = ({ id: idFromProps, query }: UseChatProps): UseChat => {
       },
       progress: -1,
     });
-  }, [setSearchQuery]);
+
+    trackEvent({
+      event_name: AnalyticsEvent.ErrorSearch,
+      target_id: id,
+    });
```

Comment:
I think this one only trigger on stream error, we should also trigger it on error we receive inside the stream.